### PR TITLE
Unit tests for azure_instances.go

### PR DESCRIPTION
**Pull request by itowlson**
**What this PR does / why we need it**: Adds unit tests to verify requests to the Azure metadata service.  These requests are used to obtain VM size, ID and address information.

**Which issue this PR fixes**: No corresponding issue

**Special notes for your reviewer**: None

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/azure/azure_test.go [hunk 1]
```diff
@@ -1051,3 +1051,160 @@ func addTestSubnet(t *testing.T, svc *v1.Service) {
...
+	}
+
+	externalIP, err := findAddress(v1.NodeExternalIP, nodeAddresses)
+	if err != nil {
+		t.Error("Expected external IP address but didn't get one")
+	}
+	if externalIP != ipAddress2 {
+		t.Errorf("Expected external IP address %s but got %s", ipAddress2, externalIP)
+	}
+
```
Comment (karataliu) [inline pkg/cloudprovider/providers/azure/azure_test.go]: extra empty line

**Comment by karataliu** [comment]
Looks like file 'azure_test.go' has grown a bit too big, and I think we can consider breaking it into pieces.
Maybe for those new tests we can add them to a new file 'azure_instances_test.go'. What do you think?

**Comment by feiskyer** [comment]
@itowlson Azure metadata has been removed in #57647. May we close this PR since all those functions are removed?

**Pull request closed without merging**